src/app-layer-detect-proto.c

```diff
@@ -634,8 +635,8 @@ static AppProto AppLayerProtoDetectPPGetProto(Flow *f,
         }
     }
 
- noparsers:
-    if (stream_config.midstream == true && idir == dir) {
+noparsers:
+    if (stream_config.midstream != false && idir == dir) {
```

Comment:
`!= false` instead of `== true`?

Reply:
Sure.
In this place, it probably does not matter. But false is exactly zero and true is exactly 1.
stream_config.midstream is declared as int.
Any nonzero value is treated as true but e.g. 123 != true literally since true is 1.
So if we are writing in C without embedded bool type then '!= false' is much more correct than '== true'

Reply:
I quick scan of the code suggests its always set to 1 or 0, so we should probably convert the type to `bool` and use it accordingly.

Reply:
Anyhow a condition "if (some_var == true)" looks confusing and dangerous in C language.
I would prefer "if (some_var != false)" or simply "if (some_var)"

I did not quite understand if you want me to apply the mentioned stream_config.midstream type change. I definitely could do that in the scope of this PR, however it makes sense only if the probing parsers change itself looks good for you.

Reply:
I think it makes sens to do the change (convert `stream_config.midstream` from int to bool) in this PR (in another commit). 

> however it makes sense only if the probing parsers change itself looks good for you.

indeed, I would like to have a suricata-verify test first

Reply:
It's already available OISF/suricata-verify#438

Reply:
Thanks, can you mention it in the PR comment
Instead of leaving `#suricata-verify-pr:`, please put `suricata-verify-pr: 438`
This way, CI uses this S-V PR to run the tests :-)